Xplat/src/main/java/vazkii/botania/common/helper/ForcePushHelper.java

```diff
@@ -0,0 +1,26 @@
+package vazkii.botania.common.helper;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ForcePushHelper {
+	private static final ThreadLocal<AtomicInteger> forcePushCounter = ThreadLocal.withInitial(AtomicInteger::new);
```

Comment:
If this is ThreadLocal, then it doesn't need to be atomic anymore (there would be no possible race conditions). Just use `Integer`?